src/main/java/nl/_42/max_login_attempts_spring_boot_starter/service/AttemptsMonitor.java

```diff
@@ -0,0 +1,46 @@
+package nl._42.max_login_attempts_spring_boot_starter.service;
+
+import java.time.Clock;
+import java.time.LocalDateTime;
+
+public class AttemptsMonitor {
+
+    private int attempts;
+    private LocalDateTime unblockTime;
+
+    public AttemptsMonitor() {
+        this.attempts = 0;
+        this.unblockTime = null;
+    }
+
+    public int getAttempts() {
+        return this.attempts;
+    }
+
+    public LocalDateTime getUnblockTime() {
+        return unblockTime;
+    }
+
+    public AttemptsMonitor addAttempt() {
+        this.attempts += 1;
+        return this;
+    }
+
+    public AttemptsMonitor block(LocalDateTime unblockTime) {
+        this.unblockTime = unblockTime;
+        return this;
+    }
+
+    public AttemptsMonitor unblock() {
+        this.unblockTime = null;
+        return this;
+    }
+
+    public AttemptsMonitor reset() {
+        return new AttemptsMonitor();
```

Comment:
Dit is tricky; je verwacht dat de interne state wordt gereset, maar krijgt een nieuwe instantie! Dan beter een static factory method `create()`. Of nog beter: de internal state resetten op deze instantie!